libs/langchain/langchain/agents/xml/base.py

```diff
@@ -97,3 +103,69 @@ async def aplan(
         }
         response = await self.llm_chain.acall(inputs, callbacks=callbacks)
         return response[self.llm_chain.output_key]
+
+
+def create_xml_agent(
+    llm: BaseLanguageModel, tools: Sequence[BaseTool], prompt: BasePromptTemplate
+) -> Runnable:
+    """Create an agent that uses XML to format its logic.
+
+    Examples:
+
+
+        .. code-block:: python
+
+            from langchain import hub
+            from langchain.chat_models import ChatAnthropic
+            from langchain.agents import AgentExecutor, create_xml_agent
+
+            prompt = hub.pull("hwchase17/xml-agent-convo")
+            model = ChatAnthropic()
+            tools = ...
+
+            agent = create_xml_agent(model, tools, prompt)
+            agent_executor = AgentExecutor(agent=agent, tools=tools)
+
+            agent_executor.invoke({"input": "hi"})
+
+            # Use with chat history
+            from langchain_core.messages import AIMessage, HumanMessage
+            agent_executor.invoke(
+                {
+                    "input": "what's my name?",
+                    # Notice that chat_history is a string
+                    # since this prompt is aimed at LLMs, not chat models
+                    "chat_history": "Human: My name is Bob\nAI: Hello Bob!",
+                }
+            )
+
+    Args:
+        llm: LLM to use as the agent.
+        tools: Tools this agent has access to.
+        prompt: The prompt to use, must have input keys of
+            `tools` and `agent_scratchpad`.
+
+    Returns:
+        A runnable sequence representing an agent. It takes as input all the same input
+        variables as the prompt passed in does. It returns as output either an
+        AgentAction or AgentFinish.
+
+    """
+    missing_vars = {"tools", "agent_scratchpad"}.difference(prompt.input_variables)
+    if missing_vars:
+        raise ValueError(f"Prompt missing required variables: {missing_vars}")
+
+    prompt = prompt.partial(
+        tools=render_text_description(list(tools)),
+    )
+    llm_with_stop = llm.bind(stop=["</tool_input>"])
```

Comment:
Add a `</final_answer>` stop sequence here?

XML output parser will split on that if it's in the response, but no reason to continue generating after that I think